applications/HDF5Application/python_scripts/core/file_io.py

```diff
@@ -0,0 +1,128 @@
+        settings.AddEmptyValue('file_name').SetString(
+            self.filename_getter.Get(identifier))
+        settings.AddEmptyValue('file_access_mode').SetString(
+            self.file_access_mode)
+        settings.AddEmptyValue('file_driver').SetString(self.file_driver)
+        settings.AddEmptyValue('echo_level').SetInt(self.echo_level)
+        return settings
+
+
+class _HDF5SerialFileIO(_FileIO):
+
+    def Get(self, identifier=None):
+        return KratosHDF5.HDF5FileSerial(self._FileSettings(identifier))
+
+
+class _HDF5ParallelFileIO(_FileIO):
+
+    def Get(self, identifier=None):
+        return KratosHDF5.HDF5FileParallel(self._FileSettings(identifier))
+
+
+class _HDF5MockFileIO(_FileIO):
+
+    def Get(self, identifier=None):
+        settings = self._FileSettings(identifier)
+        file_name = settings['file_name'].GetString()
+
+        class MockFile(object):
+            def GetFileName(self): return file_name
+        return MockFile()
+
+
+def _SetDefaults(settings):
+    default_setter = utils.DefaultSetter(settings)
+    default_setter.AddString('io_type', 'serial_hdf5_file_io')
+    default_setter.AddString('file_name', 'kratos')
+    if '<time>' in settings['file_name'].GetString():
+        default_setter.AddString('time_format', '0.4f')
+    default_setter.AddString('file_access_mode', 'exclusive')
+    if 'parallel' in settings['io_type'].GetString():
+        default_setter.AddString('file_driver', 'mpio')
+    elif os.name == 'nt':
+        default_setter.AddString('file_driver', 'windows')
+    else:
+        default_setter.AddString('file_driver', 'sec2')
+    default_setter.AddInt('echo_level', 0)
+
+
+def _GetIO(io_type):
+    if io_type == 'serial_hdf5_file_io':
+        io = _HDF5SerialFileIO()
+    elif io_type == 'parallel_hdf5_file_io':
+        io = _HDF5ParallelFileIO()
+    elif io_type == 'mock_hdf5_file_io':
+        io = _HDF5MockFileIO()
+    else:
+        raise ValueError('"io_type" has invalid value "' + io_type + '"')
+    return io
+
+
+class _FilenameGetter(object):
+
+    def __init__(self, settings):
+        filename = settings['file_name'].GetString()
+        self.filename_parts = filename.split('<time>')
+        if settings.Has('time_format'):
+            self.time_format = settings['time_format'].GetString()
+        else:
+            self.time_format = ''
+
+    def Get(self, identifier=None):
+        if hasattr(identifier, 'ProcessInfo'):
+            time = identifier.ProcessInfo[KratosMultiphysics.TIME]
+            filename = format(time, self.time_format).join(self.filename_parts)
+        else:
+            filename = ''.join(self.filename_parts)
+        if hasattr(identifier, 'Name'):
+            filename = filename.replace('<identifier>', identifier.Name)
+        if not filename.endswith('.h5'):
+            filename += '.h5'
+        return filename
+
+
+class _FilenameGetterWithDirectoryInitialization(object):
```

Comment:
This is cool, I think we could move it to the core in the long run to also use it in other things